### PR TITLE
Architecture refactoring with dependency injection

### PR DESCRIPTION
**Pull request by Pilipets**
Signed-off-by: Hlib Pylypets <pilipets.gleb@gmail.com>

### CODE REVIEW

File: web-service/src/api/controllers/controllers_common.go [hunk 1]
```diff
@@ -18,15 +19,17 @@ var (
...
 
-func InitializeControllers(container containers.ClientContainer) {
+func InitializeControllers(container containers.ClientContainer, nlp_core nlp.NlpModuleInterface) {
 	warningLogger = utils.WarningLogger
 	errorLogger = utils.ErrorLogger
 	debugLogger = utils.DebugLogger
 
 	uploadFilesDir = config.Internal.UploadFilesDir
 	db = container
+	nlpCore = nlp_core
```
Comment (gevorgyana) [inline web-service/src/api/controllers/controllers_common.go:32]: Atyle gude? No underscore

File: web-service/src/text_similarity/nlp_interface.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"bytes"
+	"fmt"
+)
+
+type PythonInternalError struct {
+	ErrMsg string
+}
+
+func NewPythonError(err_msg string) error {
```
Comment (gevorgyana) [inline web-service/src/text_similarity/nlp_interface.go:12]: Style guide control

**Review by gevorgyana — APPROVED** [review]

**Comment by Pilipets** [comment]
It's not compilable - working on it...

File: web-service/src/text_similarity/nlp_interface.go (hunk 2 above)
Comment (Pilipets) [reply web-service/src/text_similarity/nlp_interface.go:12]: Well, it comes from Python - we have some vars in camel case while others in the snake one.

File: web-service/src/api/controllers/controllers_common.go (hunk 1 above)
Comment (Pilipets) [reply web-service/src/api/controllers/controllers_common.go]: Fixed

**Review by gevorgyana — APPROVED** [review]

**Pull request merged**